train/trainer.py

```diff
@@ -219,25 +219,31 @@ def make_main_scheduler(self, optimizer):
         if self.train_params.lr_scheduler_type == 'Plateau':
             sched = torch.optim.lr_scheduler.ReduceLROnPlateau
             return sched(optimizer, mode='min', verbose=False,
-                         min_lr=self.train_params.min_lr,
-                         factor=self.train_params.scheduler_factor,
-                         patience=self.train_params.patience)
+                        min_lr=self.train_params.min_lr,
+                        factor=self.train_params.scheduler_factor,
+                        patience=self.train_params.patience)
         elif self.train_params.lr_scheduler_type == 'Cyclic':
             sched = torch.optim.lr_scheduler.CyclicLR
             return sched(optimizer, base_lr=self.train_params.min_lr,
-                         max_lr=self.train_params.lr,
-                         step_size_up=self.train_params.lr_cycle_steps // 2,
-                         mode='triangular',
-                         gamma=self.train_params.scheduler_factor,
-                         cycle_momentum=False)
+                        max_lr=self.train_params.lr,
+                        step_size_up=self.train_params.lr_cycle_steps // 2,
+                        mode='triangular',
+                        gamma=self.train_params.scheduler_factor,
+                        cycle_momentum=False)
         elif self.train_params.lr_scheduler_type == 'Cosine':
             sched = torch.optim.lr_scheduler.CosineAnnealingLR
             return sched(optimizer, self.train_params.lr_cycle_steps,
-                         eta_min=self.train_params.min_lr)
+                        eta_min=self.train_params.min_lr)
+        elif self.train_params.lr_scheduler_type == 'Linear':
+            sched = torch.optim.lr_scheduler.LinearLR
+            return sched(optimizer, start_factor=1.0,
+                        end_factor=self.train_params.min_lr / self.train_params.lr,
+                        total_iters=self.train_params.epochs * self.train_params.accumulate_grad_batches)
```

Comment:
still unclear to me here how this will operate. note that my code steps the scheduler at every training step! still not clear what accumulate_grad_batches is doing in this code